opal/runtime/opal_params_core.c

```diff
@@ -350,3 +422,73 @@ int opal_register_util_params(void)
 
     return OPAL_SUCCESS;
 }
+
+/* Parses a color 'string' and extracts the 'code' for each 'key'
+ * The string is in the format of 'key1=code1:key2=code2:...'
+ * Example: opal_var_dump_color MCA parameter
+ *
+ * color_string: The string containing the key=code sequences to parse
+ * key_names: The keys to extract (e.g. key1, key2)
+ * key_count: The length of key_names
+ * values_out: The extracted values for the keys, 1-1 with key_names
+ *   ATTENTION: Make sure it's at least as large as key_names...
+ */
+static int parse_color_string(char *color_string, char **key_names,
+        int key_count, char **values_out) {
+
+    char **tokens = opal_argv_split(color_string, ':');
+    if (NULL == tokens) {
+        return OPAL_ERR_OUT_OF_RESOURCE;
+    }
+
+    for (int i = 0; tokens[i] != NULL; i++) {
+        char **kv = opal_argv_split(tokens[i], '=');
+
+        if (NULL == kv) {
+            opal_argv_free(tokens);
+            return OPAL_ERR_OUT_OF_RESOURCE;
+        }
+
+        // Expected format of token: key=code
+        if (opal_argv_count(kv) != 2) {
+            opal_argv_free(kv);
+            continue;
+        }
+
+        // Look for key name and store value in respective position
+        for (int k = 0; k < key_count; ++k) {
+            if (strcasecmp(key_names[k], kv[0]) == 0) {
+                int ret = opal_asprintf(&values_out[k], "\033[%sm", kv[1]);
+
+                if (ret < 0) {
+                    opal_argv_free(kv);
+                    opal_argv_free(tokens);
+                    return OPAL_ERR_OUT_OF_RESOURCE;
+                }
+
+                break;
+            }
+        }
```

Comment:
Do we need to emit a show_help error here if kv[0] didn't match anything from key_names?  If not, we'll be silently ignoring a user error.